MODULE.bazel

```diff
@@ -1,6 +1,49 @@
-###############################################################################
-# Bazel now uses Bzlmod by default to manage external dependencies.
-# Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel.
-#
-# For more details, please check https://github.com/bazelbuild/bazel/issues/18958
-###############################################################################
+module(
+    name = "grpc_ecosystem_grpc_gateway",
+    version = "2.23.0",
```

Comment:
(Disclaimer: I am being super verbose about how Bazel works because I've read in some issues that you don't usually work with Bazel. Please let me know if I am being too verbose!)

Ideally it should be done with every new release. The truth is that this will not matter as much until the package is available through the [Bazel registry](https://registry.bazel.build/), because right now you need to manually specify the git hash or download the archive to make this available to other projects.

As an example, in my downstream project, this is how I would end up using this:

```starlark
# The `version` here has to match the `module` version in the project's MODULE.bazel.
bazel_dep(name = "grpc_ecosystem_grpc_gateway", version = "2.23.0")
# Tell Bazel to pull directly from the repo
git_override(
    module_name = "grpc_ecosystem_grpc_gateway",
    remote = "https://github.com/grpc-ecosystem/grpc-gateway.git",
    commit = "v2.23.0"  # This could be a any version, it's not limited to tags
)
```

As long as the version I specify in `bazel_dep` matches the version in the `module` declaration in `MODULE.bazel` this will work. It does become a little more strict when publishing to the Bazel registry though.

I believe that it would make sense to only update the `module` version when releasing, to reduce the manual burden and to better prepare the workflow for when (and if) the project is published to the Bazel registry.

Reply:
Thanks for clarifying, to be clear, it sounds like we'd need to update the `module` version _before_ tagging a new release tag in git? Is there any way we could punt that requirement until we decide to support the bazel registry down the road? If users have to override with git anyway, is it still a requirement for us to set it accurately?

Reply:
Yes, we would have to do it before tagging, which can be a bit annoying if you don't already have a release PR process defined. 

> Is there any way we could punt that requirement until we decide to support the bazel registry down the road? If users have to override with git anyway, is it still a requirement for us to set it accurately?

This comment prompted me to try to not set `version` (as it is optional) and it looks like it just works. According to the docs, not setting `version` introduces limitations when the module is depended on, but it looks like this doesn't apply if you override the dep. I am able to `bazel_dep` and `git_override` with no issues. 

An alternative approach is to use something like `version = "0.0.0"` for the time being until we introduce Bazel registry support. I think I'm going with this for the time being just because it's a little more compliant when importing the project as a module.

Reply:
Sounds good, thanks!